spec/e2e/formats_spec.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+setupTmp() { mkdir tmp; }
+cleanupTmp() { rm -rf ./tmp; }
+
+Describe 'Verifies custom rules for all formats'
```

Comment:
This test verifies the end-to-end scenario for each supported format - template, parse (if non-JSON), test, and build